We're now down to 4 test failures on Windows. These are potentially unrelated to Windows and represent prexisting bugs:
```
**********************************************************************
File "line-directive", line 222, in __main__.run
Failed example:
    output = subprocess.check_output([
       __file__, target1.name, target2.name, '--',
       sys.executable, '-c',
      'import sys;sys.stdout.write(open(sys.argv[1]).read())', raw_output.name])
Exception raised:
    Traceback (most recent call last):
      File "C:\Python27\lib\doctest.py", line 1315, in __run
        compileflags, 1) in test.globs
      File "<doctest __main__.run[12]>", line 4, in <module>
        'import sys;sys.stdout.write(open(sys.argv[1]).read())', raw_output.name])
      File "C:\Python27\lib\subprocess.py", line 567, in check_output
        process = Popen(stdout=PIPE, *popenargs, **kwargs)
      File "C:\Python27\lib\subprocess.py", line 711, in __init__
        errread, errwrite)
      File "C:\Python27\lib\subprocess.py", line 959, in _execute_child
        startupinfo)
    WindowsError: [Error 2] The system cannot find the file specified
**********************************************************************
File "line-directive", line 229, in __main__.run
Failed example:
    print output.replace(target1.name,'TARGET1-NAME').replace(target2.name,'TARGET2-NAME') + 'EOF'
Exception raised:
    Traceback (most recent call last):
      File "C:\Python27\lib\doctest.py", line 1315, in __run
        compileflags, 1) in test.globs
      File "<doctest __main__.run[13]>", line 1, in <module>
        print output.replace(target1.name,'TARGET1-NAME').replace(target2.name,'TARGET2-NAME') + 'EOF'
    NameError: name 'output' is not defined
**********************************************************************
File "line-directive", line 245, in __main__.run
Failed example:
    print subprocess.check_output([__file__, 'foo.bar', '21', target1.name]),
Exception raised:
    Traceback (most recent call last):
      File "C:\Python27\lib\doctest.py", line 1315, in __run
        compileflags, 1) in test.globs
      File "<doctest __main__.run[14]>", line 1, in <module>
        print subprocess.check_output([__file__, 'foo.bar', '21', target1.name]),
      File "C:\Python27\lib\subprocess.py", line 567, in check_output
        process = Popen(stdout=PIPE, *popenargs, **kwargs)
      File "C:\Python27\lib\subprocess.py", line 711, in __init__
        errread, errwrite)
      File "C:\Python27\lib\subprocess.py", line 959, in _execute_child
        startupinfo)
    WindowsError: [Error 2] The system cannot find the file specified
**********************************************************************
File "line-directive", line 247, in __main__.run
Failed example:
    print subprocess.check_output([__file__, 'foo.bar', '8', target2.name]),
Exception raised:
    Traceback (most recent call last):
      File "C:\Python27\lib\doctest.py", line 1315, in __run
        compileflags, 1) in test.globs
      File "<doctest __main__.run[15]>", line 1, in <module>
        print subprocess.check_output([__file__, 'foo.bar', '8', target2.name]),
      File "C:\Python27\lib\subprocess.py", line 567, in check_output
        process = Popen(stdout=PIPE, *popenargs, **kwargs)
      File "C:\Python27\lib\subprocess.py", line 711, in __init__
        errread, errwrite)
      File "C:\Python27\lib\subprocess.py", line 959, in _execute_child
        startupinfo)
    WindowsError: [Error 2] The system cannot find the file specified
**********************************************************************
1 items had failures:
   4 of  16 in __main__.run
***Test Failed*** 4 failures.

```

--- a/utils/line-directive
+++ b/utils/line-directive
@@ -61,7 +61,7 @@ def fline_map(target_filename):
 def map_line_to_source_file(target_filename, target_line_num):
     """
     >>> from tempfile import *
-    >>> t = NamedTemporaryFile()
+    >>> t = NamedTemporaryFile(delete=False)
     >>> t.write('''line 1
     ... line 2
     ... // ###sourceLocation(file: "foo.bar", line: 20)
@@ -103,7 +103,7 @@ def map_line_to_source_file(target_filename, target_line_num):
 def map_line_from_source_file(source_filename, source_line_num, target_filename):
     """
     >>> from tempfile import *
-    >>> t = NamedTemporaryFile()
+    >>> t = NamedTemporaryFile(delete=False)
     >>> t.write('''line 1
     ... line 2
     ... // ###sourceLocation(file: "foo.bar", line: 20)
@@ -173,7 +173,7 @@ def run():
     """Simulate a couple of gyb-generated files
 
     >>> from tempfile import *
-    >>> target1 = NamedTemporaryFile()
+    >>> target1 = NamedTemporaryFile(delete=False)
     >>> target1.write('''line 1
     ... line 2
     ... // ###sourceLocation(file: "foo.bar", line: 20)
@@ -184,7 +184,7 @@ def run():
     ... line 8
     ... ''')
     >>> target1.flush()
-    >>> target2 = NamedTemporaryFile()
+    >>> target2 = NamedTemporaryFile(delete=False)
     >>> target2.write('''// ###sourceLocation(file: "foo.bar", line: 7)
     ... line 2
     ... line 3
@@ -196,7 +196,7 @@ def run():
 
     Simulate the raw output of compilation
 
-    >>> raw_output = NamedTemporaryFile()
+    >>> raw_output = NamedTemporaryFile(delete=False)
     >>> target1_name, target2_name = target1.name, target2.name
     >>> raw_output.write('''A
     ... %(target1_name)s:2:111: error one

--- a/utils/line-directive
+++ b/utils/line-directive
@@ -93,6 +93,8 @@ def map_line_to_source_file(target_filename, target_line_num):
     ('baz.txt', 6)
     >>> map_line_to_source_file(t.name, 42)
     ('baz.txt', 40)
+    >>> t.close()
+    >>> os.remove(t.name)
     """
     assert(target_line_num > 0)
     map = fline_map(target_filename)
@@ -142,6 +144,8 @@ def map_line_from_source_file(source_filename, source_line_num, target_filename)
     ... except RuntimeError: pass
     >>> try: map_line_from_source_file('foo.bar', 2, t.name)
     ... except RuntimeError: pass
+    >>> t.close()
+    >>> os.remove(t.name)
     """
     assert(source_line_num > 0)
     map = fline_map(target_filename)
@@ -247,6 +251,12 @@ def run():
     >>> print subprocess.check_output([sys.executable, __file__, 'foo.bar', '8', target2.name]).replace('\\r', ''),
     3
 
+    >>> target1.close()
+    >>> os.remove(target1.name)
+    >>> target2.close()
+    >>> os.remove(target2.name)
+    >>> raw_output.close()
+    >>> os.remove(raw_output.name)
     """
     if len(sys.argv) <= 1:
         import doctest

